packages/amplify-appsync-simulator/src/velocity/util/general-utils.ts

```diff
@@ -14,10 +14,13 @@ export const generalUtils = {
     return jsStringEscape(value);
   },
   urlEncode(value) {
-    return encodeURI(value);
+    // Stringent in adhering to RFC 3986
+    return encodeURIComponent(value).replace(/[!'()*]/g, function (c) {
```

Comment:
Can you please remove the asterisk here. AppSync does not appear to encode it. I applied your changes locally and encoded `'!#$%&'()*+,/:;=?@[]'`:

Mock output: `"%21%23%24%25%26%27%28%29%2A%2B%2C%2F%3A%3B%3D%3F%40%5B%5D"`
AppSync output: `"%21%23%24%25%26%27%28%29*%2B%2C%2F%3A%3B%3D%3F%40%5B%5D"`
Mock encoding output with these changes for

Reply:
I'm not sure if you want me to move the `*` to the last place so anyone will see that we didn't forget about it. I'll leave a comment also.

Do you want me to rebase into 1 commit after is everything fine?

Reply:
> I'm not sure if you want me to move the * to the last place so anyone will see that we didn't forget about it. I'll leave a comment also.

I think a comment is enough.

> Do you want me to rebase into 1 commit after is everything fine?

Rebasing would be ideal, but we can also squash on landing.

Reply:
Sorry I meant squashing all commits into one. 

Ok. I'll leave a comment only then.